### DISCUSSION TITLE
Deployment not working! 404 Error

### DISCUSSION THREAD

**Discussion by maggiebowen** [vercel/next.js#57256] [Help]
### Summary

Hi, Thanks for your help!

I've been stuck on this problem for several weeks now. My deployment worked initially for the first few commits, but now as I continue to commit and update my main branch, I get a 404 error on my vercel site even though it works perfectly fine running on my localhost. If I run npm run build then npm run *dev it works like normal and says there are no build errors, but I get the following error when I do the command npm run start. Below is a photo of my build log.

```js
[Error: ENOENT: no such file or directory, open '/Users/annbowen/Desktop/maggie-portafolio-new/.next/BUILD_ID'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/annbowen/Desktop/maggie-portafolio-new/.next/BUILD_ID'
}
```

<img width="1058" alt="Screen Shot 2023-10-23 at 1 06 21 PM" src="https://github.com/vercel/next.js/assets/14856248/018aebf0-5f06-4206-99d6-a00fa334c3d1">

This is a link to [my vercel domain](https://maggie-portafolio-new-git-main-maggiebowen.vercel.app/) which shows a 404 error. Here's [my repository](https://github.com/maggiebowen/maggie-portafolio-new)

Please ignore the API issues I'm facing right now. The Visual Art page has an error when trying to connect to the API and fetch a JSON file, but this deployment issue existed before I encountered this new problem.

**Comment by icyJoseph** [comment]
A silly question, but have you tried, unlinking and linking again, the project to Vercel?

**Reply by maggiebowen** [reply]
Thanks for your response!

I haven't tried it yet because I've been worried it would be too hairy to unlink and link again. Would you happen to know the steps for this? I'll try it ASAP!

This is my first time using Next.js so I have only set it up as a new project from a new repository, not from an already existing repository. I'm unsure if it would be the same. Let me know!

**Reply by icyJoseph** [reply]
mmm before doing that,

if you build your project locally, and run the start command ~ does it work?

**Reply by maggiebowen** [reply]
npm run start doesn't work, just npm run dev. npm run build technically works, but its causing the 404 error when deployed. This is the error I get when running npm run start:

[Error: ENOENT: no such file or directory, open '/Users/annbowen/Desktop/maggie-portafolio-new/.next/BUILD_ID'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/annbowen/Desktop/maggie-portafolio-new/.next/BUILD_ID'
}

**Reply by icyJoseph** [reply]
Right, but you need to run `next build` first, and then `next start` immediately after.

This is not the first time I've seen people having issues with 404's on deployments though. 

I wonder if there's an issue with some of the files names, you have for example, `pages/api.js`, wouldn't that collide with `api` routes? 

In general, things that are inside `pages` folder, are an endpoint, and perhaps that's causing issues with Vercel? Not really sure.

**Reply by maggiebowen** [reply]
Yes, I ran next build and next start sequentially many times and it doesn’t
work. Before I even built out the API stuff I was getting a 404 error.
(I’ve kind of been procrastinating on this error 😅)

So perhaps I should just try to make a new project on next and deploy again
on the new project? I’m just worried I’m going to mess it up because I
haven’t done it from an existing repository. Any links and/or tips on
proceeding with this solution?

Thanks so much!!

**Reply by icyJoseph** [reply]
Ah I mean you could disconnect the project on Vercel and then reconnect it as "from GitHub". I don't know of any guides :/

**Comment by maggiebowen** [comment]
It's always opening up in localhost:3000
